### PR TITLE
nixos/matrix-synapse: make home directory configurable

### PR DESCRIPTION
**Pull request by eqyiel**
###### Motivation for this change

Just want to save my home server's data in a non-standard spot.

The default is the same as it was before so existing users won't be affected.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/services/misc/matrix-synapse.nix [hunk 1]
```diff
@@ -44,7 +44,7 @@ database: {
 }
 event_cache_size: "${cfg.event_cache_size}"
 verbose: ${cfg.verbose}
-log_file: "/var/log/matrix-synapse/homeserver.log"
+log_file: "${cfg.dataDir}/homeserver.log"
```
Comment (fpletz) [inline nixos/modules/services/misc/matrix-synapse.nix]: Do we even want a separate logfile? All logs are going to the systemd jorunal anyways.

Comment (eqyiel) [reply nixos/modules/services/misc/matrix-synapse.nix]: @fpletz I don't have strong feelings either way - should I amend this PR or make another one to remove it?

Comment (Ma27) [reply nixos/modules/services/misc/matrix-synapse.nix]: I think two separate commits for to unrealted changes should do, however I don't think that it's necessary to even open another PR :)

Comment (eqyiel) [reply nixos/modules/services/misc/matrix-synapse.nix]: @fpletz @Ma27 alright, I've added another commit that removes that line, and it seems to not have had any negative effect on my homeserver 👍

**Review by fpletz — APPROVED** [review]
Thanks for the logfile fix and for testing!

**Comment by eqyiel** [comment]
Thanks!

**Pull request merged**